packages/sign-client/src/controllers/engine.ts

```diff
@@ -148,7 +149,15 @@ export class Engine extends IEngine {
     this.isInitialized();
     await this.isValidApprove(params);
     const { id, relayProtocol, namespaces } = params;
-    const { pairingTopic, proposer, requiredNamespaces } = this.client.proposal.get(id);
+    const proposal = this.client.proposal.get(id);
+    let { pairingTopic, proposer, requiredNamespaces } = proposal;
+
+    if (!isValidObject(requiredNamespaces)) {
+      const required = getRequiredNamespacesFromNamespaces(namespaces, "approve()");
```

Comment:
it seems `getRequiredNamespacesFromNamespaces` is used only once, can `approve()` bit be hard coded into the method itself, or are you expecting this function to be used in different callers besides approve?

Reply:
Technically it can be hardcoded but set it dynamic as a precaution when/if the fx is used elsewhere as is a useful utility especially now that we're supporting empty namespaces